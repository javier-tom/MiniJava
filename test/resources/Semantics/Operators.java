class Main {
    public static void main(String[] args) {
        System.out.println(new Operators().run());
    }
}

class Operators {
    public int run() {
        int[] arr;
        int five;
        int six;
        boolean b;
        int i;
        Stuff s;

        five = 5;
        b = true;
        six = 6;
        arr = new int[six];
        i = 0;
        s = new Stuff();

        System.out.println(five - six);
        b = s.x(five + six);
        b = s.x(five * six);

        while (i < six * five + arr[0] && !b) {
            arr[i] = i;
            i = i + 1;
        }

        return arr[six - five];
    }
}

class Stuff {
    public boolean x(int black_hole) {
        return false;
    }
}
