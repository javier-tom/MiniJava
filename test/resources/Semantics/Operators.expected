class Operators:
  super: null
  fields:
  method run:
    params: 
    returns: int
    locals:
      int[] arr;
      boolean b;
      int six;
      Stuff s;
      int i;
      int five;

class Stuff:
  super: null
  fields:
  method x:
    params: int, 
    returns: boolean
    locals:

